api/views.py

```diff
@@ -1,106 +1,63 @@
-    UserLoginSerializer,
+    ProfileSerializer,
     UserRegistrationSerializer,
     UserSerializer,
     VideoSerializer,
 )
 
-JWT_PAYLOAD_HANDLER = api_settings.JWT_PAYLOAD_HANDLER
-JWT_ENCODE_HANDLER = api_settings.JWT_ENCODE_HANDLER
 
+class UserRegistrationView(generics.CreateAPIView):
+    permission_classes = (permissions.AllowAny,)
+    serializer_class = UserRegistrationSerializer
 
-class UserRegistrationView(CreateAPIView):
+    def post(self, request: request.Request):
+        response = self.create(request=request)
+        userId = response.data["id"]
+        user = User.objects.get(id=userId)
+        login(request=request, user=user)
 
-    serializer_class = UserRegistrationSerializer
-    permission_classes = (AllowAny,)
+        return response
 
-    def post(self, request):
-        serializer = self.serializer_class(data=request.data)
-        serializer.is_valid(raise_exception=True)
-        serializer.save()
-        status_code = status.HTTP_201_CREATED
-        user_profile = User.objects.get(username=request.data["username"])
 
-        payload = JWT_PAYLOAD_HANDLER(user_profile)
-        jwt_token = JWT_ENCODE_HANDLER(payload)
-        response = {
-            "success": "True",
-            "status code": status_code,
-            "message": "User registered  successfully",
-            "token": jwt_token,
-            "username": request.data["username"],
-        }
+class UserLoginView(views.APIView):
+    permission_classes = (permissions.AllowAny,)
 
-        return Response(response, status=status_code)
+    def post(self, request: request.Request):
+        username = request.data["username"]
+        password = request.data["password"]
 
+        user = authenticate(request=request, username=username, password=password)
+        if user is not None:
+            login(request=request, user=user)
+            return response.Response(status=status.HTTP_200_OK)
 
-class UserLoginView(RetrieveAPIView):
+        return response.Response(status=status.HTTP_400_BAD_REQUEST)
 
-    permission_classes = (AllowAny,)
-    serializer_class = UserLoginSerializer
 
+class UserLogoutView(generics.GenericAPIView):
     def post(self, request):
-        serializer = self.serializer_class(data=request.data)
-        serializer.is_valid(raise_exception=True)
-        response = {
-            "success": "True",
-            "status code": status.HTTP_200_OK,
-            "message": "User logged in  successfully",
-            "token": serializer.data["token"],
-            "username": request.data["username"],
-        }
-        status_code = status.HTTP_200_OK
-
-        return Response(response, status=status_code)
+        logout(request=request)
 
+        return response.Response(status=status.HTTP_200_OK)
 
-class UserProfileView(RetrieveAPIView):
 
-    permission_classes = (IsAuthenticated,)
-    authentication_class = JSONWebTokenAuthentication
+class ProfileView(generics.RetrieveAPIView):
```

Comment:
This whole view is a bit weird. It is a retrieve view but nowhere is the serializer class or queryset being used. This is effectively a regular `APIView` and the following two lines (53 and 54) serve no purpose.